execution_engine_testing/tests/src/test/chainspec_registry.rs

```diff
@@ -0,0 +1,243 @@
+use crate::lmdb_fixture;
+use casper_engine_test_support::{
+    InMemoryWasmTestBuilder, UpgradeRequestBuilder, DEFAULT_EXEC_CONFIG,
+    DEFAULT_GENESIS_CONFIG_HASH, DEFAULT_PROTOCOL_VERSION, DEFAULT_RUN_GENESIS_REQUEST,
+};
+use casper_execution_engine::core::{
+    engine_state::{run_genesis_request::RunGenesisRequest, EngineConfig, Error},
+    ChainspecRegistry, CHAINSPEC_RAW, GENESIS_ACCOUNTS_RAW, GLOBAL_STATE_RAW,
+};
+use casper_hashing::Digest;
+
+use casper_types::{EraId, Key, ProtocolVersion};
+use once_cell::sync::Lazy;
+use rand::Rng;
+
+static OLD_PROTOCOL_VERSION: Lazy<ProtocolVersion> = Lazy::new(|| *DEFAULT_PROTOCOL_VERSION);
+static NEW_PROTOCOL_VERSION: Lazy<ProtocolVersion> = Lazy::new(|| {
+    ProtocolVersion::from_parts(
+        OLD_PROTOCOL_VERSION.value().major,
+        OLD_PROTOCOL_VERSION.value().minor,
+        OLD_PROTOCOL_VERSION.value().patch + 1,
+    )
+});
+
+const DEFAULT_ACTIVATION_POINT: EraId = EraId::new(1);
+
+#[ignore]
+#[test]
+fn should_commit_chainspec_registry_during_genesis() {
+    let mut rng = rand::thread_rng();
+    let chainspec_bytes_hash = Digest::hash(rng.gen::<[u8; 32]>());
+    let genesis_account_hash = Digest::hash(rng.gen::<[u8; 32]>());
+
+    let mut chainspec_registry = ChainspecRegistry::new();
+    chainspec_registry.insert(CHAINSPEC_RAW.to_string(), chainspec_bytes_hash);
+    chainspec_registry.insert(GENESIS_ACCOUNTS_RAW.to_string(), genesis_account_hash);
+
+    let run_genesis_request = RunGenesisRequest::new(
+        *DEFAULT_GENESIS_CONFIG_HASH,
+        *DEFAULT_PROTOCOL_VERSION,
+        DEFAULT_EXEC_CONFIG.clone(),
+        chainspec_registry.clone(),
+    );
+
+    let mut builder = InMemoryWasmTestBuilder::default();
+    builder.run_genesis(&run_genesis_request);
+
+    let queried_registry = builder
+        .query(None, Key::ChainspecRegistry, &[])
+        .expect("must have entry under Key::ChainspecRegistry")
+        .as_cl_value()
+        .expect("must have underlying cl_value")
+        .to_owned()
+        .into_t::<ChainspecRegistry>()
+        .expect("must convert to chainspec registry");
+
+    let queried_chainspec_hash = queried_registry
+        .get(CHAINSPEC_RAW)
+        .expect("must have entry for chainspec_hash");
+
+    assert_eq!(*queried_chainspec_hash, chainspec_bytes_hash);
+
+    let queried_accounts_hash = queried_registry
+        .get(GENESIS_ACCOUNTS_RAW)
+        .expect("must have entry for genesis accounts");
+
+    assert_eq!(*queried_accounts_hash, genesis_account_hash);
+}
+
+#[ignore]
+#[test]
+#[should_panic]
+fn should_fail_to_commit_genesis_when_missing_chainspec_hash() {
```

Comment:
If I see correctly, this actually tests when missing both chainspec_hash and genesis accounts (i.e. chainspec registry is empty).

Since there are only two values stored in the chainspec registry now, I'd suggest testing this against all combinations (present vs. missing).

If we add more stuff to the chainspec registry in the future, we might want to move the responsibility of the validation from `GenesisInstaller::store_chainspec_registry()` to the specialized chainspec registry struct itself.